internal/services/dbaas/dbaas_resource.go

```diff
@@ -205,8 +241,51 @@ func (r *dbaasResource) Read(ctx context.Context, req resource.ReadRequest, resp
 	state.AllowedCIDRs = listFilteredIps
 	resp.Diagnostics.Append(diags...)
 
+	oldStateEffectiveConfiguration := make(map[string]string)
+	resp.Diagnostics.Append(state.EffectiveConfiguration.ElementsAs(ctx, &oldStateEffectiveConfiguration, false)...)
+
+	state.EffectiveConfiguration, resp.Diagnostics = r.refreshEffectiveConfiguration(
+		ctx,
+		state.PublicCloudId.ValueInt64(),
+		state.PublicCloudProjectId.ValueInt64(),
+		state.Id.ValueInt64(),
+		&resp.Diagnostics,
+	)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !state.Configuration.IsNull() {
+		stateConfiguration := make(map[string]string)
+		resp.Diagnostics.Append(state.Configuration.ElementsAs(ctx, &stateConfiguration, false)...)
+
+		stateEffectiveConfiguration := make(map[string]string)
+		resp.Diagnostics.Append(state.EffectiveConfiguration.ElementsAs(ctx, &stateEffectiveConfiguration, false)...)
+
+		for effectiveConfigurationKey, effectiveConfigurationValue := range stateEffectiveConfiguration {
+			_, ok := stateConfiguration[effectiveConfigurationKey]
+			if ok {
+				stateConfiguration[effectiveConfigurationKey] = effectiveConfigurationValue
+			}
+
+			oldEffectiveConfigurationValue, okOld := oldStateEffectiveConfiguration[effectiveConfigurationKey]
+			stateEffectiveConfigValue, ok := stateEffectiveConfiguration[effectiveConfigurationKey]
+			if okOld && (!ok || stateEffectiveConfigValue != oldEffectiveConfigurationValue) {
+				stateConfiguration[effectiveConfigurationKey] = effectiveConfigurationValue
+			}
+		}
+
+		mapStateConfiguration, diags := types.MapValueFrom(ctx, types.StringType, stateConfiguration)
+		state.Configuration = mapStateConfiguration
+		resp.Diagnostics.Append(diags...)
+	}
```

Comment:
This part seems rather complex, should we make it a function and add a little bit of documentation ?